### DISCUSSION TITLE
Metamask + Wallet Connect 2 Auth - no response

### DISCUSSION THREAD

**Discussion by itsallmememe** [WalletConnect/walletconnect-monorepo#2811] [Developer Support] [answered]
I’m creating a native iOS dApp that uses MetaMask and Rainbow for Auth and signing transactions using Wallet Connect 2, but I’m running into some difficulty. All the following code works with Rainbow.

I setup as follows. The DefaultSocketFactory and DefaultCryptoProvider are from the demo app

```
        Networking.configure(projectId: Self.ProjectId, socketFactory: DefaultSocketFactory())
        Auth.configure(crypto: DefaultCryptoProvider())

        let metadata = AppMetadata(name: "Source",
                                   description: "A mobile application acting as a Dapp to connect to a crypto wallet using the WalletConnect SDK",
                                   url: Self.ClientUrl.absoluteString,
                                   icons: ["https://i.ebayimg.com/images/g/5qgAAOSwoBtW3zvq/s-l400.jpg"])

        Pair.configure(metadata: metadata)

        Task {
            try await Sign.instance.cleanup()
        }
```

Then I auth using this code

```
    func authenticate(usingWallet wallet: WalletConnectType) {
        Task(priority: .userInitiated) {
            guard let uri else {
                Task { @MainActor in
                    self.authenticatedState = .notAuthenticated
                }
                return
            }

            do {
                try await Auth.instance.request(.stub(), topic: uri.topic)
            } catch {
                Task { @MainActor in
                    self.authenticatedState = .notAuthenticated
                }
            }

            Task { @MainActor in
                let url = URL(string: "https://\(wallet.universalLinkHost)/wc?uri=\(uri.deeplinkUri)")
                UIApplication.shared.open(url!)
            }
        }
    }

private extension RequestParams {
    static func stub(
        domain: String = "service.invalid",
        chainId: String = "eip155:1",
        nonce: String = "32891756",
        aud: String = "https://service.invalid/login",
        nbf: String? = nil,
        exp: String? = nil,
        statement: String? = "I accept the ServiceOrg Terms of Service: https://service.invalid/tos",
        requestId: String? = nil,
        resources: [String]? = ["ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/", "https://example.com/my-web2-claim.json"]
    ) -> RequestParams {
        return RequestParams(
            domain: domain,
            chainId: chainId,
            nonce: nonce,
            aud: aud,
            nbf: nbf,
            exp: exp,
            statement: statement,
            requestId: requestId,
            resources: resources
        )
    }
}
```

MetaMask opens, but it does not react. Testing with Rainbow, it opens and shows the authentication popup and auths as expected.

Any ideas what I'm doing wrong?

**Comment by adrianotadao** [comment]
@itsallmememe check this out https://github.com/rainbow-me/rainbowkit/issues/1262#issuecomment-1601598463

Everybody seems to be waiting for this fix

**Reply by itsallmememe** [reply]
@adrianotadao - I don't think this is appropriate. My app is opening up the native Metamask app directly. 
I am saying that the code above works on Rainbow, but not metamask. Therefore it has nothing to do with rainbowkit.

**Reply by adrianotadao** [reply]
@itsallmememe I was giving you a context of the conversation. What matters here is the link to this https://github.com/MetaMask/metamask-mobile/issues/6457

**Reply by itsallmememe** [reply]
@adrianotadao thanks for the clarification! 
Metamask opens for the deeplinks and does react to the Sign.connect request, but it does not communicate back to the app in that instance. And an Auth (and I assume Sign.request) are just ignored... so the deeplink is recognised just in very specific circumstances.

**Reply by itsallmememe** [reply]
@adrianotadao - can you upvote this issue please? Might help it get looked at quicker.

**Reply by adrianotadao** [reply]
@itsallmememe done!

**Reply by itsallmememe** [reply]
@adrianotadao - I've posted on the Metamask forums, can you upvote this ticket too please? https://github.com/MetaMask/metamask-mobile/issues/6873

**Comment by SunZhiC** [comment]
Any update on this issue, I meet the same bug with WalletConnectSwiftV2, app jump to MetaMask and show connect succeed, but there is no response from MetaMask.
And it works with Trust, Bitkeep, Imtoken and other wallets.

**Reply by itsallmememe** [reply]
@SunZhiC I haven't found a solution yet...

**Reply by itsallmememe** [reply]
I have also posted in the metamask dev support but not heard anything back - https://community.metamask.io/t/wallet-connect-2-auth-no-response/26146

**Reply by itsallmememe** [reply]
@SunZhiC  - can you upvote this issue please? Might help it get looked at quicker.

**Reply by itsallmememe** [reply]
@SunZhiC  - I've posted on the Metamask forums, can you upvote this ticket too please? https://github.com/MetaMask/metamask-mobile/issues/6873

**Comment by dibuenio** [comment]
Hi guys,
Were you able to fix this problem? I have the same. It seems that contacting technical support will not do any good. It is very important for the launch of our product and users, but we are very dependent on solving problems with your product.

**Reply by itsallmememe** [reply]
@dibuenio  - can you upvote this issue please? Might help it get looked at quicker.

**Reply by itsallmememe** [reply]
@dibuenio I've posted on the Metamask forums, can you upvote this ticket too please? https://github.com/MetaMask/metamask-mobile/issues/6873

**Comment by finessevanes** [comment]
Hey everyone - thanks for bumping this. Could you please provide the version of the SDK where you are experiencing these problems?

**Reply by itsallmememe** [reply]
Hi @finessevanes - thanks for replying. I'm using 1.6.6. Again, it works ok with Rainbow but not Metamask.

**Reply by SunZhiC** [reply]
1.6.10, it works with Trust, Bitkeep, Imtoken, but not Metamask, I think its Metamask's bug.

**Reply by itsallmememe** [reply]
@finessevanes - any movement on this? We're launching soon and it's really holding us back

**Reply by itsallmememe** [reply]
@finessevanes sorry to bump, we're launching very soon and this is a major hurdle for us.

**Comment by maltsevoff** [comment, marked as answer]
metamask team has answered to this issue that this isn't a part of WCv2 feature and closed it. 
https://github.com/MetaMask/metamask-mobile/issues/6704

**Comment by pqseags** [comment]
@maltsevoff Do you know if Metamask supports Sign.connect? We're seeing similar issues there: https://github.com/MetaMask/metamask-mobile/issues/6908
